pkgs/by-name/sh/shark/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  boost,
+  cmake,
+  fetchFromGitHub,
+  openssl,
+  stdenv,
+}:
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "shark";
+    version = "4.1.0";
```

Comment:
This is not an official release

Reply:
Thanks @SuperSandro2000 !

> May 21, 2024: We made some updates to Shark, and if you you clone the repository, it should work nicely. You might call this Shark 4.1.0, however, we will not do official Shark releases anymore. We will try to maintain the current status of the library. Please simply checkout the most recent versions from the GitHub repository.

Shark will not have any more official releases. [c.f](https://shark-ml.github.io/Shark/) will be just maintained, as they mentioned in readme so I named it as 4.1.0. 

This package also is specific to OTB, do you recommend this to be moved inside OTB as well ?

Reply:
> You might call this Shark 4.1.0, however, we will not do official Shark releases anymore.

Not sure what they are trying to achieve but it is stupid.

Then we must use the unstable format. https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming

Reply:
Thanks @SuperSandro2000 !

Last release version was `4.0.1` and the latest commit was on `2024-05-25` so based on that, would it be correct to name the version as  `4.0.1-unstable-2024-05-25` ?

Reply:
I have named it as `4.0-unstable-2024-05-25`  as the last release was 4.0.1.